### PR TITLE
Added recently opened files list

### PR DESCRIPTION
**Pull request by CPKreuz**
Can be found under File > Recent

### CODE REVIEW

**Comment by CPKreuz** [comment]
(╯°□°）╯︵ ┻━┻

File: PixiEditor/Models/DataHolders/Document/Document.IO.cs [hunk 1]
```diff
@@ -47,5 +49,32 @@ public void Save(string path)
...
+            {
+                recentlyOpened.Insert(0, newPath);
+            }
+            else
+            {
+                int index = recentlyOpened.IndexOf(newPath);
+                recentlyOpened.Move(index, 0);
+            }
+
+            if (recentlyOpened.Count > 5)
```
Comment (flabbet) [inline PixiEditor/Models/DataHolders/Document/Document.IO.cs:67]: Make this a setting in preferences and set the default to 10

File: PixiEditor/Models/UserPreferences/PreferencesSettings.cs [hunk 2]
```diff
@@ -107,5 +118,54 @@ public static void AddCallback(string setting, Action<object> action)
...
+        }
+
+#nullable disable
+
+        private static string GetPathToSettings(Environment.SpecialFolder folder)
+        {
+            return Path.Join(
+            Environment.GetFolderPath(folder),
+            "PixiEditor",
+            "user_preferences.json");
```
Comment (flabbet) [inline PixiEditor/Models/UserPreferences/PreferencesSettings.cs:146]: This is a minor thing, but for settings that are generated by the editor, I would change the name to `editor_data.json` or something like that

File: PixiEditor/Views/Dialogs/NoticePopup.xaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <TextBlock Grid.Row="1" Text="{Binding Body, ElementName=popup}" HorizontalAlignment="Center"
+                   VerticalAlignment="Center" FontSize="18" Foreground="White" />
+        <DockPanel Grid.Row="0" Background="{StaticResource MainColor}">
+            <Button DockPanel.Dock="Right" HorizontalAlignment="Right" Style="{StaticResource CloseButtonStyle}"
+                    WindowChrome.IsHitTestVisibleInChrome="True" ToolTip="Close"
+                    Command="{Binding DataContext.CancelCommand, ElementName=popup}" />
+        </DockPanel>
+        <StackPanel Grid.Row="2" Orientation="Horizontal" VerticalAlignment="Bottom" HorizontalAlignment="Center"
+                    Margin="0,0,10,10">
+            <Button Height="30" Width="60"
```
Comment (flabbet) [inline PixiEditor/Views/Dialogs/NoticePopup.xaml:34]: Check ConfirmationDialog for reference, the button should be aligned to right, not in middle. Let's be consistent

File: PixiEditor/Views/Dialogs/NoticePopup.xaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                    Click="Close"
```
Comment (flabbet) [inline PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: Also what does this Click do

File: PixiEditor/Views/MainWindow.xaml [hunk 5]
```diff
@@ -83,6 +83,14 @@
                 <MenuItem Header="_File">
                     <MenuItem InputGestureText="CTRL+N" Header="_New" Command="{Binding FileSubViewModel.OpenNewFilePopupCommand}" />
                     <MenuItem Header="_Open" InputGestureText="Ctrl+O" Command="{Binding FileSubViewModel.OpenFileCommand}" />
+                    <MenuItem Header="_Recent" ItemsSource="{Binding FileSubViewModel.RecentlyOpened}" x:Name="recentItemMenu" IsEnabled="{Binding FileSubViewModel.HasRecent}">
```
Comment (flabbet) [inline PixiEditor/Views/MainWindow.xaml:86]: Change blue hover to match rest of menu items.

**Review by flabbet — CHANGES_REQUESTED** [review]
Also would be awesome if you wrote tests, for example for new Preference things you added.

File: PixiEditor/Views/Dialogs/NoticePopup.xaml (hunk 4 above)
Comment (CPKreuz) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: Close the window?

Comment (flabbet) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: But it's a string, not a handler. How does this work

Comment (CPKreuz) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.buttonbase.click?view=net-5.0

Comment (flabbet) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: ok nvm, I was confused by the name

Comment (flabbet) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:35]: change this to OkButton_Click

**Comment by CPKreuz** [comment]
> Also would be awesome if you wrote tests, for example for new Preference things you added.

Can I write a test for ViewModelMain?

File: PixiEditor/Views/Dialogs/NoticePopup.xaml (hunk 3 above)
Comment (CPKreuz) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:34]: But it's in the middle in `ConfirmationDialog`

Comment (flabbet) [reply PixiEditor/Views/Dialogs/NoticePopup.xaml:34]: oh, nvm

**Comment by flabbet** [comment]
Sure, what will you test?

**Comment by CPKreuz** [comment]
> Sure, what will you test?

If the path gets added to recently opened when setting the document path

**Comment by CPKreuz** [comment]
I would say it's a nice achievement when you manage to break the whole application even though all tests are passing

**Review by flabbet — CHANGES_REQUESTED** [review]
Fixed crashes, btw, recent files doesn't work for non .pixi files. It does show in menu, but it crashes, since program tries to open it as .pixi, not .png

**Review by flabbet — CHANGES_REQUESTED** [review]
You can open the same recent file twice at the same time. It shouldn't be like that

**Comment by CPKreuz** [comment]
> You can open the same recent file twice at the same time. It shouldn't be like that

Multitasking!

**Review by flabbet — APPROVED** [review]
I'll approve, but please write tests next time

**Pull request merged**